src/handlers/folder.rs

```diff
@@ -45,12 +46,10 @@ use crate::services::FolderService;
 )]
 pub async fn get(
     State(ctx): State<Arc<Context>>,
-
-    Path(id): Path<Uuid>,
-    Path(reference): Path<String>,
-    Path(path): Path<String>,
+    Path((id, reference, path)): Path<(Uuid, String, Option<String>)>,
+    Query(params): Query<HashMap<String, String>>,
 ) -> Result<impl IntoResponse> {
-    Ok(Json(FolderService::get(ctx, id, reference, path).await?))
+    Ok(Json(FolderService::get(ctx, id, reference, path, params).await?))
```

Comment:
Pass all params if the api need more than one params, else just pass recursive, like this: params.get("recursive")